xc/device.ml

```diff
@@ -1983,30 +1983,52 @@ module Dm_Common = struct
     ; fd_map = []
     }
 
+  let nvidia_compat_lookup config_file =
+    let rec find = function
+      | [] -> raise Not_found
+      | conf :: others ->
+        match String.split_on_char ' ' conf with
+        | [cf; type_id] when cf = config_file -> type_id
+        | _ -> find others
+    in
+    try
+      Unixext.string_of_file !Xenopsd.nvidia_compat_lookup_file
+      |> String.split_on_char '\n'
+      |> find
+    with e ->
+      error "nvidia_compat_lookup failed with %s" (Printexc.to_string e);
```

Comment:
Might be worth logging the config_file that we failed to look up to help with debugging.

Reply:
Yep, I've added a fixup.